db/mysql/ops.go

```diff
@@ -166,10 +166,26 @@ func scanChatRooms(rows *sql.Rows) ([]*domain.ChatRoom, error) {
 
 	for rows.Next() {
 		var chatRoom domain.ChatRoom
-		if err := rows.Scan(&chatRoom.ID, &chatRoom.Name); err != nil {
+		var userId string
+
+		if err := rows.Scan(&chatRoom.ID, &chatRoom.Name, &userId); err != nil {
 			return nil, fmt.Errorf("failed to scan chatroom row: %w", err)
 		}
-		chatRooms = append(chatRooms, &chatRoom)
+
+		if len(chatRooms) == 0 {
+			chatRoom.Users = append(chatRoom.Users, userId)
+			chatRooms = append(chatRooms, &chatRoom)
+		} else {
+			for i := 0; i < len(chatRooms); i++ {
+				if chatRoom.ID == chatRooms[i].ID {
+					chatRoom.Users = append(chatRoom.Users, userId)
+				} else {
+					chatRoom.Users = append(chatRoom.Users, userId)
+					chatRooms = append(chatRooms, &chatRoom)
+					break
+				}
+			}
+		}
 	}
```

Comment:
this does the job but its not efficient. can you optimize this logic?

Reply:
Using a map now. Also, removed the nested loop